pkg/client/client.go

```diff
@@ -93,10 +93,9 @@ func NewClient(ctx context.Context, hostname string, port int, certFile string,
 		grpc.WithTransportCredentials(creds),
 
 		// grpc.WithInsecure(),
-		grpc.WithBlock(),
 	}
 	// Set up a connection to the server.
-	conn, err := grpc.DialContext(ctx, address, opts...)
+	conn, err := grpc.NewClient(address, opts...)
```

Comment:
As this is no longer blocking i will add a Version() endpoint for clients to be able to wait until the service is available.

Reply:
Maybe we can even add this to health checks of the metal-api then?

Reply:
voted against it because the masterdata-api does not start if the database is not working as required.